### ISSUE TITLE
interleave-reads.py fails with zero exit status: ERROR: This doesn't look like paired data!

### ISSUE THREAD

**Issue by fungs** [dib-lab/khmer#1853]
Hi, I discovered by pure luck, because it created an empty output file, that interleave-reads.py created incomplete output with zero exit status.

version: 2.1.2
executable: interleave-reads.py
call: interleave-reads.py in1.fq in2.fq > out.fq

The reason why it failed is because reads where contamination-filtered and quality-trimmed with BBMap suite before, which changed the length=299 in the Illumina read identifiers to have different lengths in a read pair. This doesn't make it invalid input. I'll file another issue for that.

Best,
Johannes

**Comment by ctb** [comment]
Hi @fungs could you provide us with example stderr output here? I am looking at the source code and the exit status should be nonzero in all error conditions - trying to figure out what error condition you encountered that didn't trigger it! and thank you for catching all of this!

**Comment by fungs** [comment]
I hope I didn't oversee anything, but I ran this with GNU parallel which showed all calls to return with 0. The error message I saw on the terminal is in the title:

ERROR: This doesn't look like paired data! [followed by the read deflines]

Best,
Johannes

**Comment by fungs** [comment]
I see the sys.exit(1) in the code. I investigated my command, again, and think I have found the problem on my side. Sorry for the false alarm. This is about how BASH handles exit codes and pipes.

I compressed stdout using

    interleave-reads.py .. | pigz > ...

If the first command fails, the exit code is still zero. I didn't consider that.

    set -o pipefail
    interleave-reads.py .. | pigz > ...

Triggers the right behavior. I will keep that in mind.

Best,
Johannes
